src/MongoDB.Driver.Core/Core/WireProtocol/CommandUsingQueryMessageWireProtocol.cs

```diff
@@ -51,27 +55,37 @@ internal class CommandUsingQueryMessageWireProtocol<TCommandResult> : IWireProto
             ReadPreference readPreference,
             DatabaseNamespace databaseNamespace,
             BsonDocument command,
+            IEnumerable<Type1CommandMessageSection> commandPayloads,
             IElementNameValidator commandValidator,
             BsonDocument additionalOptions,
-            Func<CommandResponseHandling> responseHandling,
+            CommandResponseHandling responseHandling,
             IBsonSerializer<TCommandResult> resultSerializer,
-            MessageEncoderSettings messageEncoderSettings)
+            MessageEncoderSettings messageEncoderSettings,
+            Action<IMessageEncoderPostProcessor> postWriteAction)
         {
+            if (responseHandling != CommandResponseHandling.Return && responseHandling != CommandResponseHandling.Ignore)
+            {
+                throw new ArgumentException("CommandResponseHandling must be Return or Ignore.", nameof(responseHandling));
+            }
+
             _session = Ensure.IsNotNull(session, nameof(session));
             _readPreference = readPreference;
             _databaseNamespace = Ensure.IsNotNull(databaseNamespace, nameof(databaseNamespace));
             _command = Ensure.IsNotNull(command, nameof(command));
+            _commandPayloads = commandPayloads?.ToList(); // can be null
             _commandValidator = Ensure.IsNotNull(commandValidator, nameof(commandValidator));
             _additionalOptions = additionalOptions; // can be null
             _responseHandling = responseHandling;
             _resultSerializer = Ensure.IsNotNull(resultSerializer, nameof(resultSerializer));
             _messageEncoderSettings = messageEncoderSettings;
+            _postWriteAction = postWriteAction; // can be null
         }
 
         // methods
         private QueryMessage CreateMessage(ConnectionDescription connectionDescription, out bool messageContainsSessionId)
         {
-            var wrappedCommand = WrapCommandForQueryMessage(connectionDescription, out messageContainsSessionId);
+            var commandWithPayloads = CombineCommandWithPayloads(connectionDescription);
```

Comment:
This is where we call a helper method to insert the payloads into the command itself since OP_QUERY doesn't support payloads.